### PR TITLE
Add ability to track distributions

### PR DESCRIPTION
**Pull request by mihran113**
- Add `Distribution`: `CustomObject` sub-class
 - Add `Distributions`: `Sequence` sub-class
 - Add utility functions for distributions and allow querying for distributions
 - Add endpoints and utils for visualization of distributions on single run page view
 - Some refactoring along the way

### CODE REVIEW

**Comment by mihran113** [comment]
Resolves #1060

File: aim/storage/treeutils_.pyx [hunk 1]
```diff
@@ -1,3 +1,6 @@
+from typing import Any, Iterator, Tuple, Union
+from copy import deepcopy
```
Comment (alberttorosyan) [inline aim/storage/treeutils_.pyx:2]: seems this import ins not used

File: aim/sdk/distribution_sequence.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        return self._meta_tree['first_step']
+
+    def last_step(self):
+        """Get sequence tracked last step.
+
+        Required to implement ranged and sliced data fetching.
+        """
+        return self._meta_tree['last_step']
+
+    def record_length(self):
```
Comment (alberttorosyan) [inline aim/sdk/distribution_sequence.py:32]: do we need this method?

**Review by alberttorosyan — COMMENTED** [review]
looks good. just need to fix a couple of minor things.

**Comment by mihran113** [comment]
> looks good. just need to fix a couple of minor things.

Updated based on the comments, please have a look

**Review by alberttorosyan — APPROVED** [review]
lgtm

**Pull request merged**